corrections10/17

jhdalek55 opened the pull request:
I made the corrections that definitely need fixing. Will follow-up with e-mail on a few things that maybe should be changed.

jhdalek55 committed:
corrections10/17

dgopstein commented on about.md:
```diff
@@ -17,5 +17,4 @@ We are a team of researchers at several universities working to uncover the funa
 
-We can be reached at atomsofconfusion at googlegroups
-
+We can be reached at atomsofconfusion@googlegroups.com
```
This was intentionally left not as an email address so we don't get spammed

jhdalek55 replied:
Ok. It should be back the way it was now.

dgopstein replied:
Did you push your changes? I can't see the update here

dgopstein commented on publications.md:
```diff
@@ -10,5 +10,4 @@ The following is a list of publications coming from this project:
 
-> Humans often mistake the meaning of source code, and so misjudge a program’s true behavior. To better under- stand code patterns that confuse programmers, we extracted a preliminary set from winners of the International Obfuscated C Code Contest (IOCCC). We show experimentally (using 73 participants) that most of these code patterns lead to a signifi- cantly increased rate of misunderstanding v ...
-
+> Humans often mistake the meaning of source code, and so misjudge a program’s true behavior. To better understand code patterns that confuse programmers, we extracted a preliminary set from winners of the International Obfuscated C Code Contest (IOCCC). We show experimentally (using 73 participants) that most of these code patterns lead to a significantly increased rate of misunderstanding versu ...
```
What did you change here? This was the abstract from the paper.

jhdalek55 replied:
There were some odd breaks in a few words. That was all I changed here, I believe.

dgopstein reviewed (commented):
These changes look great. Please revert the email address and I'll pull.

Thanks!

jhdalek55 commented:
On Oct 17, 2016 6:16 PM, "Dan Gopstein" notifications@github.com wrote:

> _@dgopstein_ commented on this pull request.
> 
> These changes look great. Please revert the email address and I'll pull.
> 
> ## Thanks!
> 
> In about.md
> https://github.com/AtomsofConfusion/AtomsofConfusion.github.io/pull/2#pullrequestreview-4569168
> :
> 
> > @@ -17,5 +17,4 @@ We are a team of researchers at several universities working to uncover the funa
> 
>  ### Contact
> 
> ## -We can be reached at atomsofconfusion at googlegroups
> 
> +We can be reached at atomsofconfusion@googlegroups.com
> 
> ## This was intentionally left not as an email address so we don't get spammed
> 
> In publications.md
> https://github.com/AtomsofConfusion/AtomsofConfusion.github.io/pull/2#pullrequestreview-4569168
> :
> 
> > @@ -10,5 +10,4 @@ The following is a list of publications coming from this project:
> >  D. Gopstein, J. Iannacone, Y. Yan, L. DeLong, Y. Zhuang, M. Yeh, J. Cappos<br />
> >  _Under Review_
> 
> ## -> Humans often mistake the meaning of source code, and so misjudge a program’s true behavior. To better under- stand code patterns that confuse programmers, we extracted a preliminary set from winners of the International Obfuscated C Code Contest (IOCCC). We show experimentally (using 73 participants) that most of these code patterns lead to a signifi- cantly increased rate of misunderstanding versus equivalent code without the pattern. We go on to empirically demonstrate (using 43 participants) that removing the design patterns that were validated as confusing from IOCCC winners has a significant effect on the level of comprehension. These studies provide not only a rigorous foundation to support software engineering best practices, but also resulted in several new findings. This study shows that extremely small patterns of code can have a very large impact on code comprehension. Surprisingly, the level of confusion an individual programmer has when reading these design patterns varies widely in ways that do not correlate with the programmer’s experience or education. We also found evidence that popular style guidelines are inclusive of some confusing patterns, while prohibiting others that do not have a statistically significant effect on confusion.
> 
> +> Humans often mistake the meaning of source code, and so misjudge a program’s true behavior. To better understand code patterns that confuse programmers, we extracted a preliminary set from winners of the International Obfuscated C Code Contest (IOCCC). We show experimentally (using 73 participants) that most of these code patterns lead to a significantly increased rate of misunderstanding versus equivalent code without the pattern. We go on to empirically demonstrate (using 43 participants) that removing the design patterns that were validated as confusing from IOCCC winners has a significant effect on the level of comprehension. These studies provide not only a rigorous foundation to support software engineering best practices, but also resulted in several new findings. This study shows that extremely small patterns of code can have a very large impact on code comprehension. Surprisingly, the level of confusion an individual programmer has when reading these design patterns varies widely in ways that do not correlate with the programmer’s experience or education. We also found evidence that popular style guidelines are inclusive of some confusing patterns, while prohibiting others that do not have a statistically significant effect on confusion.
> 
> What did you change here? This was the abstract from the paper.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/AtomsofConfusion/AtomsofConfusion.github.io/pull/2#pullrequestreview-4569168,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AN5qX_QFp9HItjSf-Krq8T299v38Hml7ks5q0_PYgaJpZM4KY0Lv
> .
> 
> Sorry about that. Should have made that a question rather than an edit.

Would it be possible to do this tomorrow?  I'm not really set up to fix
this at home. In terms of the abstract, I believe all I was fix some odd
breaks in words.

Lois

dgopstein commented:
I'm in no hurry whatsoever, work on this when it's convenient for you. Also, don't feel bad about proposing a fix that ultimately get's rejected, that's exactly what the pull-request process is designed to encourage. I think what we're doing now is good.

jhdalek55 committed:
fixed .com

Hope I did this right

jhdalek55 commented:
I have issues w/git. I went back to the branch, changed it and saved 
 it. Maybe it went as a separate commit?
Lois

On Tue, Oct 18, 2016 at 10:54 AM, Dan Gopstein wrote:

@dgopstein commented on this pull request.

In about.md 
https://github.com/AtomsofConfusion/AtomsofConfusion.github.io/pull/2 
:

> @@ -17,5 +17,4 @@ We are a team of researchers at several universities 
> working to uncover the funa

  ### Contact

## -We can be reached at atomsofconfusion at googlegroups

+We can be reached at atomsofconfusion@googlegroups.com

Did you push your changes? I can't see the update here

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub 
https://github.com/AtomsofConfusion/AtomsofConfusion.github.io/pull/2 
, or mute the thread 
https://github.com/notifications/unsubscribe-auth/AN5qXwjASPQ37ys14np8ZaW1uRZGYpSVks5q1N3BgaJpZM4KY0Lv 
.

jhdalek55 committed:
.com

jhdalek55 committed:
Remove old contact page and fix email address in about
